### ISSUE TITLE
Consistency: Should we include `creator` into the software_name annotation?

### ISSUE THREAD

**Issue by caifand** [howisonlab/softcite-dataset#641]
I am moving some existing issues into standalone posts to increase their visibility.

The first one is what we've debated for some time. For `software_name` annotations like *Microsoft Excel*, *GraphPad Prism*, "Lotus Notes*, should we exclude the creator name from the `software_name`? Apart from their semantic difference and the introduced ambiguities, one big concern brought up by @kermitt2 earlier is to avoid overlapping annotations since they will become knotty in tei xml.

Currently in our dataset, *GraphPad Prism* are usually put together in `software_name`. In some cases *Microsoft* is separately annotated as `creator` while the corresponding `software_name` is annotated as *Excel*; but we also have tricky examples like *MS*+*Excel*/*MS Excel*/"Microsoft*+*Office Excel*, etc. e.g.:

`<p>All statistical analyses were performed using paired Student's t tests and <rs corresp="#PMC3025493-software-3" type="creator">Microsoft</rs> <rs type="software" xml:id="PMC3025493-software-3">Excel</rs> or <rs type="software">Prism</rs> software packages.

Calculations were made using <rs type="software">MS Excel</rs> and are presented in Appendix 1.

used to summarise the analytic outputs using <rs corresp="#PMC5435264-software-0" type="creator">MS</rs>
          <rs type="software" xml:id="PMC5435264-software-0">Excel</rs>.`

Sometimes there's additional `creator` info accompanied.

`Observed heterozygosity was estimated in Microsoft <rs type="software" xml:id="PMC4103605-software-13">Excel</rs> (<rs corresp="#PMC4103605-software-13" type="creator">Microsoft Corporation</rs>, Redmond, Washington, USA).
`

If we strictly limit only one annotated string in each annotation field, then we would want to sort the cases with both *Microsoft* occurs before and after the *Excel*. We can set a rule here for establishing some priority (e.g., annotate the full organizational name and ignore the *Microsoft* before the software name) when annotating. The same as the case of *GraphPad Prism*. 

Generally speaking, I think it's reasonable to separate "GraphPad" from *Prism* and do the same thing to *Microsoft*/*MS* *Excel*.  Perhaps *IBM Notes* is a hypothesized example as we only have one instance of *Lotus Notes* in the candidate tei xml.

**Comment by jameshowison** [comment]
So the rule should be:

`software_name` never includes a preceding publisher.

If there is a "instrument-like" citation following the `software_name` then we code creator in there but not preceding publishers. (e.g., 
```MS <rs type="software_name">Excel</rs> (<rs type="creator">Microsoft Corporation</rs>, Redmond, WA)

```
<rs type="software_name">Excel</rs> by <rs type="creator">Microsoft<rs>
``` 

Otherwise we code creator in the preceding publisher.

```
Calculations were made using <rs type="creator">MS</rs> <rs type="software">Excel</rs>.
```
```
We used <rs type="creator">GraphPad<rs> <rs type="software">Prism</rs>.
```

**Comment by kermitt2** [comment]
This is indeed exactly the rules I tried to follow for having some consistency - except for the raised cases like _GraphPad_ Prism and _Lotus Notes_ for which the "publisher" name is so commonly attached to the actual software name that it's only after reviewing many paragraphs that I realized that the rule was not applied. 

I think it makes sense however to apply the rules systematically, so having 

```
We used <rs type="creator">GraphPad<rs> <rs type="software">Prism</rs>.
```

and 

```
<rs type="creator">Lotus</rs> <rs type="software">Notes</rs>
```

**Comment by kermitt2** [comment]
I have to confess also a bias :) 

I think I kept those few exceptions like `Lotus Notes`, because I had in mind the problem of disambiguation/matching of the software mention in existing software knowledge bases. I know that after extracting all software name mention, we want to deduplicate them and match them to a software "entity". 

If you look at the "labels" for the Wikidata entity for _Lotus Notes_, at https://www.wikidata.org/wiki/Q60198 you see that they all contain the publisher name. So having this bias helps the matching, not having `Lotus` and just `Notes` make the matching a bit more complicated as we need to combine different extracted fields.

**Comment by jameshowison** [comment]
Yeah, I mean the reality is that some software has the publisher in the
name, and even the publisher uses that.  I definitely see that.  But we
need some sort of consistency here, no?
